jax/_src/lax/control_flow/loops.py

```diff
@@ -1029,6 +1031,59 @@ def _scan_pp_rule(eqn, context, settings):
     del printed_params['reverse']
   return core._pp_eqn(eqn.replace(params=printed_params), context, settings)
 
+def _scan_discharge_rule(in_avals, out_avals, *args, jaxpr, num_consts,
+                         num_carry, linear, unroll, reverse, length):
+  jaxpr, consts = jaxpr.jaxpr, jaxpr.consts
+  if consts: raise NotImplementedError
+  consts, carry, xs = split_list(args, [num_consts, num_carry])
+  consts_linear, carry_linear, xs_linear = split_list(
+      linear, [num_consts, num_carry])
+  consts_avals, carry_avals, xs_avals = split_list(in_avals,
+                                                   [num_consts, num_carry])
+  is_ref = [isinstance(a, state.AbstractRef) for a in consts_avals]
+  remaining_const_avals, in_ref_avals = partition_list(is_ref, consts_avals)
+  remaining_consts, in_refs = partition_list(is_ref, consts)
+  remaining_consts_linear, in_refs_linear = partition_list(is_ref, consts_linear)
+  num_refs = sum(is_ref)
+  num_extensive_in = len(in_avals) - num_carry - num_consts
+  num_extensive_out = len(out_avals) - num_carry
+  num_remaining_consts = num_consts - num_refs
+  discharged_jaxpr, discharged_consts = state_discharge.discharge_state(jaxpr, ())
+  if discharged_consts: raise NotImplementedError
```

Comment:
This would only happen if a discharge rule had consts in it? Maybe we should add a message like "If you see this, open an issue at https://github.com/google/jax/issues".

Reply:
Done.